tests/runner/src/main/java/org/mule/test/runner/api/AetherClassPathClassifier.java

```diff
@@ -268,20 +322,73 @@ private PluginUrlClassification buildPluginUrlClassification(Artifact pluginArti
       throw new IllegalStateException("Couldn't resolve dependencies for plugin: '" + pluginArtifact + "' classification", e);
     }
 
-    if (context.isExtensionMetadataGenerationEnabled()) {
-      File generatedMetadataFolder = extensionPluginGenerator.generateExtensionManifest(pluginArtifact, urls);
-      if (generatedMetadataFolder != null) {
-        URL generatedTestResources = toUrl(generatedMetadataFolder);
+    List<Dependency> directDependencies;
```

Comment:
Can you extract utility methods on all these big methods? Is very hard to understand what a method does when there are too many things going on